cfsan-snp-pipeline/2.2.1/Dockerfile

```diff
@@ -0,0 +1,121 @@
+	&& python get-pip.py
+
+#install bowtie2
+#add -std:c++11 to make for compiler support
+RUN wget https://github.com/BenLangmead/bowtie2/archive/v$BOWTIE2_VER.tar.gz -qO						 			- | tar xz && cd bowtie2-$BOWTIE2_VER  && make && make install && cd /tmp
+
+#install samtools
+RUN	wget https://github.com/samtools/htslib/releases/download/$HTSLIB_VER/htslib-$HTSLIB_VER.tar.bz2 -qO       		- | tar xj && (cd htslib-$HTSLIB_VER  	 && make && make install && cd /tmp)
+RUN wget https://github.com/samtools/samtools/releases/download/$SAMTOOLS_VER/samtools-$SAMTOOLS_VER.tar.bz2 -qO  	- | tar xj && (cd samtools-$SAMTOOLS_VER && make && make install && cd /tmp)
+RUN wget https://github.com/samtools/bcftools/releases/download/$BCFTOOLS_VER/bcftools-$BCFTOOLS_VER.tar.bz2 -qO 	- | tar xj && (cd bcftools-$BCFTOOLS_VER && make && make install && cd /tmp)
+
+#install VARSCAN, ART, SRA Toolkit, GATK, Picard
+RUN wget http://downloads.sourceforge.net/project/varscan/VarScan.v$VARSCAN_VER.jar -q \
+	&& cp VarScan.v$VARSCAN_VER.jar /usr/bin/VarScan.jar 
+RUN wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/$SRATOOLKIT_VER/sratoolkit.$SRATOOLKIT_VER-ubuntu64.tar.gz -q \
+	&& tar -zxf /tmp/sratoolkit.$SRATOOLKIT_VER-ubuntu64.tar.gz \
+	&& cp /tmp/sratoolkit.$SRATOOLKIT_VER-ubuntu64/bin/fastq-dump.$SRATOOLKIT_VER /usr/bin/fastq-dump 
+RUN wget https://storage.googleapis.com/gatk-software/package-archive/gatk/GenomeAnalysisTK-$GATK_VER.tar.bz2 \
+	&& tar -jxf /tmp/GenomeAnalysisTK-$GATK_VER.tar.bz2 \
+	&& cp /tmp/GenomeAnalysisTK.jar /usr/bin/GenomeAnalysisTK.jar 
+RUN wget https://github.com/broadinstitute/picard/releases/download/$PICARD_VER/picard.jar -q \
+	&& cp picard.jar /usr/bin/picard.jar
+
+#install snp-pipeline and snp-mutator
+RUN pip install numpy==$NUMPY_VER \
+	snp-pipeline==$SNP_PIPELINE_VER \
+	biopython==$BIOPYTHON_VER
+
+
+ENV PATH "$PATH:/tmp/samtools-$SAMTOOLS_VER/bin:/tmp/bcftools-$BCFTOOLS_VER/bin:/tmp/bowtie2-$BOWTIE2_VER/bin"
+ENV CLASSPATH "/usr/bin/VarScan.jar:/usr/bin/picard.jar:/usr/bin/GenomeAnalysisTK.jar"
+ENV NUMCORES 1
+
+# make a DIR called /data for running cfsan-snp-pipeline in
+WORKDIR /data
+
+# Test snp_pipeline (code modified from Justin). Running copy_snppipeline_data.py generates an message the the script is deprecated
+# but I figured I would leave it since the below commands test that the image runs properly.
+FROM app as test
+WORKDIR /test/
+RUN cfsan_snp_pipeline data lambdaVirusInputs testLambdaVirus \
+	&& cd testLambdaVirus \
+	&& cfsan_snp_pipeline run -s samples reference/lambda_virus.fasta \
+	&& copy_snppipeline_data.py lambdaVirusExpectedResults expectedResults \
+	&& diff -q snplist.txt expectedResults/snplist.txt >> diffOutput.txt \
+	&& diff -q snpma.fasta expectedResults/snpma.fasta >> diffOutput.txt \
+	&& diff -q referenceSNP.fasta expectedResults/referenceSNP.fasta >> diffOutput.txt \
+    && if [ -s diffOutput.txt ]; then echo "testLambdaVirus failed"; else echo "testLambdaVirus passed"; fi
```

Comment:
I like this test:
- uses cfsan_snp_pipeline to download some test data
- runs several cfsan_snp_pipeline functions
- checks created files against known outpus